libyul/optimiser/CommonSubexpressionEliminator.cpp

```diff
@@ -96,10 +108,17 @@ void CommonSubexpressionEliminator::visit(Expression& _e)
 	}
 	else
 	{
+		// TODO lifetime?
+		static Literal zero{{}, LiteralKind::Number, "0"_yulstring, {}};
 		// TODO this search is rather inefficient.
 		for (auto const& [variable, value]: m_value)
 		{
 			assertThrow(value.value, OptimizerException, "");
+			if (
+				m_returnVariables.top().count(variable) &&
+				SyntacticallyEqual{}(*value.value, zero)
```

Comment:
This one I mean. Why this? Isn't this a *bad* thing to do? Or well, it doesn't cost at runtime, but code size will increase needlessly... or do we expect a new code transform to generally track all values and ``dup`` whenever it knows a value is on stack?

Reply:
Might actually not be the worst of ideas to have the code transform alone take care of reusing literals... but I'm a bit afraid that if we just move everything to the responsibility of the code transform, it'll become a monster...

Reply:
But if we go that way we probably shouldn't even restrict this to zeroes, but extend it to everything that's a literal?
Going that way would probably also mean that the code transform would either need to *also* do the constant optimization or be able to evaluate what it did...

Reply:
This  does not replace a zero by a return variable. Nothing else.

Reply:
Yeah, nevermind, I somehow read an "or", not an "and", sorry.